app/controllers/HomePageController.scala

```diff
@@ -83,44 +82,61 @@ class HomePageController @Inject() (
+
+    val pageLoadFuture: Future[Result] =
+      for {
+        _                                       <- sessionService.resetAll()
+        // Get the available count of biks available for each tax year
+        biksListOptionCY: List[Bik]             <-
+          bikListService.registeredBenefitsList(controllersReferenceData.yearRange.cyminus1, EmpRef("", ""))(
+            uriInformation.getBenefitTypesPath
+          )
+        biksListOptionCYP1: List[Bik]           <-
+          bikListService.registeredBenefitsList(controllersReferenceData.yearRange.cy, EmpRef("", ""))(
+            uriInformation.getBenefitTypesPath
+          )
+        currentYearList: List[Bik]              <- bikListService.currentYearList
+        nextYearListResponse: HttpResponse      <- bikListService.getNextYearList
+        nextYearList                             = nextYearListResponse.json.as[List[Bik]]
+        nextYearListHeaders: Map[String, String] = Map(
+                                                     HeaderTags.ETAG   -> nextYearListResponse
+                                                       .header(HeaderTags.ETAG)
+                                                       .getOrElse("0"),
+                                                     HeaderTags.X_TXID -> nextYearListResponse
+                                                       .header(HeaderTags.X_TXID)
+                                                       .getOrElse("1")
+                                                   )
+      } yield {
+        sessionService.storeCYRegisteredBiks(currentYearList)
+        sessionService.storeNYRegisteredBiks(nextYearList)
+
+        val fromYTA = if (request.session.get(ControllersReferenceDataCodes.SESSION_FROM_YTA).isDefined) {
+          request.session.get(ControllersReferenceDataCodes.SESSION_FROM_YTA).get
+        } else {
+          isFromYTA
+        }
+        auditHomePageView()
+        Ok(
+          summaryPage(
+            pbikAppConfig.cyEnabled,
+            taxYearRange,
+            currentYearList,
+            nextYearList,
+            biksListOptionCY.size,
+            biksListOptionCYP1.size,
+            fromYTA.toString,
+            empRef = request.empRef
+          )
         )
-      currentYearList: (Map[String, String], List[Bik]) <- bikListService.currentYearList
-      nextYearList: (Map[String, String], List[Bik])    <- bikListService.nextYearList
-    } yield {
-      sessionService.storeCYRegisteredBiks(currentYearList._2)
-      sessionService.storeNYRegisteredBiks(nextYearList._2)
-      val fromYTA = if (request.session.get(ControllersReferenceDataCodes.SESSION_FROM_YTA).isDefined) {
-        request.session.get(ControllersReferenceDataCodes.SESSION_FROM_YTA).get
-      } else {
-        isFromYTA
+          .addingToSession(nextYearListHeaders.toSeq: _*)
```

Comment:
preserve adding to session of the headers but this time not pulled or stored from a Map found in the HMRCTierConnector.